libraries/ssh_known_hosts.rb

```diff
@@ -0,0 +1,147 @@
+#!/usr/bin/env ruby
+
+module SshKnownHosts
+  # Describes a host (IP, domain) for a host entry
+  class Host
+    @regex_ip_address = Regexp.new('\d+\.\d+\.\d+\.\d+')
+
+    def self.from_string(str)
+      ips = []
+      domains = []
+      str.split(',').each do |part|
+        if @regex_ip_address.match(part)
+          ips.push(part)
+        else
+          domains.push(part)
+        end
+      end
+      new(domains, ips)
+    end
+
+    def initialize(domains, ips)
+      @domains = domains.uniq.sort
+      @ips = ips.uniq.sort
+    end
+
+    attr_reader :domains
+
+    attr_reader :ips
+
+    def to_s
+      host = domains + ips
+      host.join(',')
+    end
+  end
+
+  # Host key class for a value object describing a known_host key (type key)
+  class HostKey
+    def self.from_parts(key, type)
+      type = "ssh-#{type}" if type == 'rsa' || type == 'dsa'
```

Comment:
he seems to have reduced the options to only `rsa`
https://github.com/shrikeh/chef-ssh-user-known-hosts/pull/3/files#diff-2337bdead6fc65d0020bb3221b0be0bfR60should we implement (now) all of the other types or we could leave that for later?